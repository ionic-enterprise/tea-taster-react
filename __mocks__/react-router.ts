import { vi } from 'vitest';

const replace = vi.fn();
const useHistory = vi.fn().mockReturnValue({ replace });

export { useHistory };
