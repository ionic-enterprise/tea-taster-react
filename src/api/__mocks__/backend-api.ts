import { vi } from 'vitest';

const client = {
  get: vi.fn().mockResolvedValue({ data: null }),
  post: vi.fn().mockResolvedValue({ data: null }),
  delete: vi.fn().mockResolvedValue({ data: null }),
};

export { client };
