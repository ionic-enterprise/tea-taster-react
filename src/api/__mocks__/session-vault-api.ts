import { vi } from 'vitest';

const canUnlock = vi.fn().mockResolvedValue(undefined);
const getUnlockMode = vi.fn().mockResolvedValue(undefined);
const setUnlockMode = vi.fn().mockResolvedValue(undefined);
const clearSession = vi.fn().mockResolvedValue(undefined);
const getSession = vi.fn().mockResolvedValue(undefined);
const setSession = vi.fn().mockResolvedValue(undefined);

export { canUnlock, getUnlockMode, setUnlockMode, clearSession, getSession, setSession };
