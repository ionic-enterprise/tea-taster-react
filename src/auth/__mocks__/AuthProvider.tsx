import { vi } from 'vitest';

const login = vi.fn().mockResolvedValue(undefined);
const logout = vi.fn().mockResolvedValue(undefined);

export const useAuth = () => ({ login, logout });
