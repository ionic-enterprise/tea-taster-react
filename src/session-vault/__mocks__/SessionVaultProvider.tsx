import { vi } from 'vitest';

const canUnlock = vi.fn().mockResolvedValue(undefined);
const getUnlockMode = vi.fn().mockResolvedValue(undefined);
const setUnlockMode = vi.fn().mockResolvedValue(undefined);

export const useSessionVault = () => ({ canUnlock, getUnlockMode, setUnlockMode });

const SessionVaultProvider = ({ children }: any) => <div>{children}</div>;
export default SessionVaultProvider;
