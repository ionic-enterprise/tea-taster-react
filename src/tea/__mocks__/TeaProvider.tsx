import { vi } from 'vitest';

const rate = vi.fn().mockResolvedValue(undefined);
export const useTea = vi.fn(() => ({ teas: [], rate }));
