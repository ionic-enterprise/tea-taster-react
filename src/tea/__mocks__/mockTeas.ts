import { Tea } from '../../shared/models';

export const mockTeas: Array<Tea> = [
  {
    id: 1,
    name: 'Green',
    image: require('../../assets/images/green.jpg'),
    description:
      'Green teas have the oxidation process stopped very early on, leaving them with a very subtle flavor and ' +
      'complex undertones. These teas should be steeped at lower temperatures for shorter periods of time.',
  },
  {
    id: 2,
    name: 'Black',
    image: require('../../assets/images/black.jpg'),
    description:
      'A fully oxidized tea, black teas have a dark color and a full robust and pronounced flavor. Black teas tend ' +
      'to have a higher caffeine content than other teas.',
  },
  {
    id: 3,
    name: 'Herbal',
    image: require('../../assets/images/herbal.jpg'),
    description:
      'Herbal infusions are not actually "tea" but are more accurately characterized as infused beverages ' +
      'consisting of various dried herbs, spices, and fruits.',
  },
  {
    id: 4,
    name: 'Oolong',
    image: require('../../assets/images/oolong.jpg'),
    description:
      'Oolong teas are partially oxidized, giving them a flavor that is not as robust as black teas but also ' +
      'not as suble as green teas. Oolong teas often have a flowery fragrance.',
  },
  {
    id: 5,
    name: 'Dark',
    image: require('../../assets/images/dark.jpg'),
    description:
      'From the Hunan and Sichuan provinces of China, dark teas are flavorful aged probiotic teas that steeps ' +
      'up very smooth with slightly sweet notes.',
  },
  {
    id: 6,
    name: 'Puer',
    image: require('../../assets/images/puer.jpg'),
    description:
      'An aged black tea from china. Puer teas have a strong rich flavor that could be described as "woody" or "peaty."',
  },
  {
    id: 7,
    name: 'White',
    image: require('../../assets/images/white.jpg'),
    description:
      'White tea is produced using very young shoots with no oxidation process. White tea has an extremely ' +
      'delicate flavor that is sweet and fragrent. White tea should be steeped at lower temperatures for ' +
      'short periods of time.',
  },
  {
    id: 8,
    name: 'Yellow',
    image: require('../../assets/images/yellow.jpg'),
    description:
      'A rare tea from China, yellow tea goes through a similar shortened oxidation process like green teas. ' +
      'Yellow teas, however, do not have the grassy flavor that green teas tend to have. The leaves often ' +
      'resemble the shoots of white teas, but are slightly oxidized.',
  },
];

export const resultTeas = () => {
  return expectedTeas.map((t: Tea) => {
    const tea = { ...t };
    delete tea.image;
    delete tea.rating;
    return tea;
  });
};

const expectedTeas = [
  {
    id: 1,
    name: 'Green',
    image: 'green.jpg',
    description: 'Green tea description.',
    rating: 1,
  },
  {
    id: 2,
    name: 'Black',
    image: 'black.jpg',
    description: 'Black tea description.',
    rating: 2,
  },
  {
    id: 3,
    name: 'Herbal',
    image: 'herbal.jpg',
    description: 'Herbal Infusion description.',
    rating: 3,
  },
  {
    id: 4,
    name: 'Oolong',
    image: 'oolong.jpg',
    description: 'Oolong tea description.',
    rating: 4,
  },
  {
    id: 5,
    name: 'Dark',
    image: 'dark.jpg',
    description: 'Dark tea description.',
    rating: 5,
  },
  {
    id: 6,
    name: 'Puer',
    image: 'puer.jpg',
    description: 'Puer tea description.',
    rating: 0,
  },
  {
    id: 7,
    name: 'White',
    image: 'white.jpg',
    description: 'White tea description.',
    rating: 0,
  },
  {
    id: 8,
    name: 'Yellow',
    image: 'yellow.jpg',
    description: 'Yellow tea description.',
    rating: 0,
  },
];
