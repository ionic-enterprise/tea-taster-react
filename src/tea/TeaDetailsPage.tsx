import { IonBackButton, IonButtons, IonContent, IonHeader, IonImg, IonPage, IonTitle, IonToolbar } from '@ionic/react';
import { useTea } from './TeaProvider';
import { useParams } from 'react-router';
import { Tea } from '../models';
import { Rating } from '../shared/Rating';
import { useState } from 'react';

const TeaDetailsPage: React.FC = () => {
  const { id } = useParams<{ id: string }>();
  const { teas } = useTea();
  const tea: Tea | undefined = teas.find((t) => t.id === parseInt(id, 10));

  const [rating, setRating] = useState<number>(2);

  return (
    <IonPage>
      <IonHeader>
        <IonToolbar>
          <IonButtons slot="start">
            <IonBackButton defaultHref="/tea" />
          </IonButtons>
          <IonTitle>Tea Details</IonTitle>
        </IonToolbar>
      </IonHeader>
      <IonContent className="ion-padding">
        {tea && (
          <div>
            <div className="ion-justify-content-center" style={{ display: 'flex' }}>
              <IonImg src={tea.image} style={{ maxWidth: '75%', maxHeight: '512px' }} />
            </div>
            <h1 data-testid="name">{tea.name}</h1>
            <p data-testid="description">{tea.description}</p>
            <Rating rating={rating} onRatingChange={(r) => setRating(r)} />
          </div>
        )}
      </IonContent>
    </IonPage>
  );
};
export default TeaDetailsPage;
