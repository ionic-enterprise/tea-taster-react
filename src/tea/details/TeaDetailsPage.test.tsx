import React from 'react';
import { waitForIonicReact } from '@ionic/react-test-utils';
import { render, waitFor } from '@testing-library/react';
import TeaDetailsPage from './TeaDetailsPage';
import { expectedTeas } from '../__mocks__/mockTeas';

jest.mock('react-router', () => ({
  useParams: () => ({
    id: '1',
  }),
}));

const mockTea = expectedTeas[0];
jest.mock('../useTea', () => ({
  useTea: () => ({
    getTeas: jest.fn(),
    getTeaById: jest.fn(() => Promise.resolve(mockTea)),
  }),
}));

describe('<TeaDetailsPage />', () => {
  it('displays the header', async () => {
    const { container } = render(<TeaDetailsPage />);
    await waitFor(() => waitForIonicReact());
    await waitFor(() => expect(container).toHaveTextContent(/Details/));
  });

  it('renders consistently', async () => {
    const { asFragment, container } = render(<TeaDetailsPage />);
    await waitFor(() => waitForIonicReact());
    await waitFor(() => expect(container).toHaveTextContent(mockTea.description));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the tea name', async () => {
    const { container } = render(<TeaDetailsPage />);
    await waitFor(() => waitForIonicReact());
    await waitFor(() => expect(container).toHaveTextContent(mockTea.name));
  });

  it('renders the tea description', async () => {
    const { container } = render(<TeaDetailsPage />);
    await waitFor(() => waitForIonicReact());
    await waitFor(() => expect(container).toHaveTextContent(mockTea.description));
  });
});
