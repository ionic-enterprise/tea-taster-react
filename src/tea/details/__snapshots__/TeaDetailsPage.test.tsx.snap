// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeaDetailsPage /> renders consistently 1`] = `
<DocumentFragment>
  <div
    class="ion-page"
  >
    <ion-header
      class="md header-md header-collapse-none"
      role="banner"
    >
      <!---->
      <ion-toolbar
        class="toolbar-title-default md in-toolbar"
      >
        <ion-buttons
          class="buttons-first-slot sc-ion-buttons-md-h sc-ion-buttons-md-s md"
          slot="start"
        >
          <ion-back-button
            class="md button back-button-has-icon-only in-toolbar ion-activatable ion-focusable show-back-button"
            default-href="/tabs/tea"
            defaulthref="/tabs/tea"
          />
        </ion-buttons>
        <ion-title
          class="md title-default"
        >
          Details
        </ion-title>
      </ion-toolbar>
    </ion-header>
    <ion-content
      class="main-content md"
      style="--offset-top: 0px; --offset-bottom: 0px;"
    >
      <ion-header
        class="md header-md header-collapse-condense"
        collapse="condense"
        role="banner"
      >
        <!---->
        <ion-toolbar
          class="toolbar-title-large md in-toolbar"
        >
          <ion-title
            class="md title-large"
            size="large"
          >
            Details
          </ion-title>
        </ion-toolbar>
      </ion-header>
      <div
        class="ion-padding"
      >
        <div
          class="ion-justify-content-center"
        >
          <ion-img
            class="md"
          />
        </div>
        <h1>
          Green
        </h1>
        <p>
          Green tea description.
        </p>
        <div
          class="rating"
          style="opacity: 1;"
        >
          <ion-icon
            class="md"
            data-testid="Rate 1 stars"
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
            role="img"
          />
          <ion-icon
            class="md"
            data-testid="Rate 2 stars"
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
            role="img"
          />
          <ion-icon
            class="md"
            data-testid="Rate 3 stars"
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
            role="img"
          />
          <ion-icon
            class="md"
            data-testid="Rate 4 stars"
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
            role="img"
          />
          <ion-icon
            class="md"
            data-testid="Rate 5 stars"
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
            role="img"
          />
        </div>
      </div>
    </ion-content>
  </div>
</DocumentFragment>
`;
